sys/net/routing/rpl/of0.c

```diff
@@ -75,19 +75,35 @@ uint16_t calc_rank(rpl_parent_t *parent, uint16_t base_rank)
     return base_rank + add;
 }
 
-/* We simply return the Parent with lower rank */
+/**
+ * @brief Return the parent with lower rank.
+ *
+ * @param[in] p1 parent used as left element in the relation
+ * @param[in] p2 parent used as right element in the relation
+ * @returns the parent with the lower rank or p1 if the rank of p1 and p2 are equal
+ */
 rpl_parent_t *which_parent(rpl_parent_t *p1, rpl_parent_t *p2)
 {
-    if (p1->rank < p2->rank) {
+    if (p1->rank <= p2->rank) {
```

Comment:
Don't know if it would make sense to additionally check if the passed pointers are `!= NULL`, since they are only used here [1] and must have been initialized in advance.
Is there a situation where this function could be used somewhere else?

[1] https://github.com/RIOT-OS/RIOT/blob/master/sys/net/routing/rpl/rpl_dodag.c#L274